src/cli/rustup_mode.rs

```diff
@@ -925,12 +916,17 @@ async fn check_updates(cfg: &Cfg<'_>, opts: CheckOpts) -> Result<ExitCode> {
     // Check for update only if rustup does **not** have the no-self-update feature,
     // and auto-self-update is configured to **enable**
     // and has **no** no-self-update parameter.
-    let self_update = !cfg!(feature = "no-self-update")
-        && self_update_mode == SelfUpdateMode::Enable
-        && !opts.no_self_update;
+    let self_update =
+        CFG_SELF_UPDATE && self_update_mode == SelfUpdateMode::Enable && !opts.no_self_update;
 
-    if self_update && check_rustup_update(&DownloadCfg::new(cfg)).await? {
-        update_available = true;
+    if self_update {
+        // Print the separating newline only if toolchain updates were found.
+        if update_available {
```

Comment:
@djc Is that the right place to add a newline, or doesn't the multi-progress bar subsume a newline at the end?